benchmark/framework/agency.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This test module contains AEA/AEABuilder wrapper to make performance tests easy."""
+from threading import Thread
+from typing import Dict, List, Optional, Tuple, Type, Union
+
+from aea.aea import AEA
+from aea.aea_builder import AEABuilder
+from aea.configurations.base import ProtocolId, SkillConfig
+from aea.configurations.components import Component
+from aea.crypto.fetchai import FETCHAI
+from aea.mail.base import Envelope
+from aea.protocols.base import Message
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.skills.base import Handler, Skill, SkillContext
+
+
+class Agency:
+    """A testing wrapper to run and control an agent."""
+
+    def __init__(self, name: str = "my_eae", skills: List[dict] = None):
+        """Make an agency with optional name and skills."""
+        self.skills = skills or []
+        self.name = name
+
+        self.agent = self.make_agent(
+            self.name, [self.make_skill(**skill) for skill in self.skills]
+        )
+
+    def make_agent(
+        self, name: str = "my_eae", components: List[Component] = None
+    ) -> AEA:
+        """
+        Create AEA from name and already loaded components.
+
+        :return: AEA
+        """
+        components = components or []
+        builder = AEABuilder()
+
+        builder.set_name("my_aea")
+        builder.add_private_key(FETCHAI, "")
+
+        for component in components:
+            builder._resources.add_component(component)
```

Comment:
why are you not using `builder.add_component(component)`?

Reply:
In particular, the access to the private resources should not be necessary here.

Reply:
Especially, as this bypasses the consistency checks.

Reply:
this is a test wrapper, the main idea does not follow all the building patterns from the builder, but create a test environment. not sure builder has all required methods to manage resources programmatically way